From 5deec2e4f3bc3e21f7b8fb65f9f75276be0920d0 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Tue, 23 Nov 2010 13:54:02 +0800
Subject: [PATCH] ENGR00133955 MX50: dma apbh incorrect ioremap size

The ioremap for dma apbh registers mapping is not correct.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arch/arm/mach-mx5/dma-apbh.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx5/dma-apbh.c b/arch/arm/mach-mx5/dma-apbh.c
index d617561..9e1b10d 100644
--- a/arch/arm/mach-mx5/dma-apbh.c
+++ b/arch/arm/mach-mx5/dma-apbh.c
@@ -166,7 +166,7 @@ static int __devinit dma_apbh_probe(struct platform_device *pdev)
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	if (!res)
 		return -ENOMEM;
-	mxs_dma_apbh.base = ioremap(res->start, res->end);
+	mxs_dma_apbh.base = ioremap(res->start, resource_size(res));
 	__raw_writel(BM_APBH_CTRL0_SFTRST,
 		     mxs_dma_apbh.base + HW_APBH_CTRL0_CLR);
 	for (i = 0; i < 10000; i++) {
-- 
1.5.4.4

